src/full/Agda/TypeChecking/Monad/Base.hs

```diff
@@ -3893,6 +3967,41 @@ aModeToDef AbstractMode = Just AbstractDef
 aModeToDef ConcreteMode = Just ConcreteDef
 aModeToDef _ = Nothing
 
+---------------------------------------------------------------------------
+-- ** Opaque blocks
+---------------------------------------------------------------------------
+
+-- | A block of opaque definitions.
+data OpaqueBlock = OpaqueBlock
+  { opaqueId        :: {-# UNPACK #-} !OpaqueId
+    -- ^ Unique identifier for this block.
+  , opaqueUnfolding :: HashSet QName
```

Comment:
Do we need `HashSet` here?  Why not just `Set`?  Isn't this a premature optimization?
Note that `HashSet.toList` gives you different results depending of which version of `hashable` you compiled against.

Reply:
This map is used very heavily (every single getConstInfo when in an opaque block...), I'd rather not deoptimise this.

Using a different type with the same interface is not premature optimisation IMO: that'd have been my original idea of also having a Bloom filter.